src/libsyntax_pos/symbol.rs

```diff
@@ -384,8 +384,6 @@ impl<'a> ::std::cmp::PartialEq<InternedString> for &'a String {
     }
```

Comment:
How is this sound?

Reply:
Oh, `InternedString` storing a `&'static str` isn't sound at all. I'll have to change it to using a `u32` like Symbol.

Reply:
Well that made it unable to implement `Deref<Target=str>`, so I went with the approach suggested in a FIXME and just leak interned strings.